api/v1beta1/disruption_types.go

```diff
@@ -317,3 +317,43 @@ func ReadUnmarshal(path string) (*Disruption, error) {
 
 	return &parsedSpec, nil
 }
+
+// RemoveDeadTargets removes targets not found in matchingTargets from the targets list
+func (status *DisruptionStatus) RemoveDeadTargets(matchingTargets []string) {
+	for index := 0; index < len(status.Targets); index++ {
+		if !contains(matchingTargets, status.Targets[index]) {
+			status.Targets[len(status.Targets)-1], status.Targets[index] = status.Targets[index], status.Targets[len(status.Targets)-1]
+			status.Targets = status.Targets[:len(status.Targets)-1]
+		}
+	}
+}
+
+// AddTargets adds newTargetsCount random targets from the eligibleTargets list to the Target List
+func (status *DisruptionStatus) AddTargets(newTargetsCount int, eligibleTargets []string) {
+	for i := 0; i < newTargetsCount; i++ {
+		index := rand.Intn(len(eligibleTargets)) //nolint:gosec
+		status.Targets = append(status.Targets, eligibleTargets[index])
+		eligibleTargets[len(eligibleTargets)-1], eligibleTargets[index] = eligibleTargets[index], eligibleTargets[len(eligibleTargets)-1]
+		eligibleTargets = eligibleTargets[:len(eligibleTargets)-1]
+	}
+}
+
+// RemoveTargets removes toRemoveTargetsCount random targets from the Target List
+func (status *DisruptionStatus) RemoveTargets(toRemoveTargetsCount int) {
+	for i := 0; i < toRemoveTargetsCount; i++ {
+		index := rand.Intn(len(status.Targets)) //nolint:gosec
+		status.Targets[len(status.Targets)-1], status.Targets[index] = status.Targets[index], status.Targets[len(status.Targets)-1]
+		status.Targets = status.Targets[:len(status.Targets)-1]
+	}
+}
+
+// contains returns true when the given string is present in the given slice
+func contains(s []string, str string) bool {
```

Comment:
This is a re-implement of the function in `controller/helpers.go` right? Could we extract from both places, and move to some sort of util or helpers package?

Reply:
Sounds like a fair package to create

Reply:
Created an `utils` package with only this for now
I can imagine with time we can move various things there